misc-scripts/gifts_import/populate_protein_feature.pl

```diff
@@ -0,0 +1,350 @@
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#The script populates the CORE db's protein_feature table with all the uniprot mappings found in GIFTS db's ensp_u_cigar table.
+
+
+use strict;
+use warnings;
+use DBI qw( :sql_types );
+use Getopt::Long;
+use Bio::EnsEMBL::Registry;
+use Data::Dumper;
+
+my ( $gfhost, $gfport, $gfuser, $gfpass, $gfdb );
+my ( $pfhost, $pfport, $pfuser, $pfpass, $pfdb );
+
+my $create_index = 0;
+my $dryrun = 0;
+#my $limit = 10;
+my $limit = 0;
+
+
+GetOptions(
+  "gfhost=s"    => \$gfhost,
+  "gfport=i"       => \$gfport,
+  "gfuser=s"    => \$gfuser,
+  "gfpass=s"    => \$gfpass,
+  "gfdb=s"   => \$gfdb,
+
+  "pfhost=s"      => \$pfhost,
+  "pfport=i"      => \$pfport,
+  "pfuser=s"      => \$pfuser,
+  "pfpass=s"        => \$pfpass,
+  "pfdb=s"   => \$pfdb,
+
+  "dryrun!"         => \$dryrun,
+  "help", \&usage,
+
+);
+
+usage() if ( !defined $gfhost || !defined $gfport || !defined $gfuser || !defined $gfdb );
+usage() if ( !defined $pfhost  || !defined $pfport || !defined $pfuser || !defined $pfpass );
+
+
+my ( $giftsDB, $protfeatureDB ) = init_db();
+die unless defined($giftsDB) and defined($protfeatureDB);
+
+print "giftsDB credentials:\n";
+print Dumper($giftsDB);
+
+print "protfeatureDB credentials:\n";
+print Dumper($protfeatureDB);
+
+
+my $registry = get_registry();
+my $translation_adaptor = $registry->get_adaptor( 'human', 'core', 'translation' );
+
+
+sub init_db {
+
+# This is where the gifts info will be read from
+  my $giftsDB = {
+    user => $gfuser,
+    pass => $gfpass,
+    host => $gfhost,
+    port => $gfport,
+    dbname => $gfdb
+
+  };
+
+# This is where the gifts info will be written to
+  my $protfeatureDB = {
+    user   => $pfuser,
+    pass   => $pfpass,
+    host   => $pfhost,
+    port   => $pfport,
+    dbname => $pfdb
+  };
+
+  return ( $giftsDB, $protfeatureDB );
+}
+
+
+populate_protein_feature_db();
+
+sub get_analysis_id {
+  my ($logic_name) = @_;
+  my $analysis_id;
+  
+  my $dba = get_dba($protfeatureDB);
+  my $analysis_dba = $dba->get_AnalysisAdaptor();
+  my $analysis = $analysis_dba->fetch_by_logic_name($logic_name);
+  
+  $analysis_id = $analysis->dbID() if defined $analysis;
+  return $analysis_id;
+}
+
+
+sub populate_protein_feature_db {
+
+  my $gifts_dbh = db_connect($giftsDB );
+  my $protfeature_dbh = db_connect($protfeatureDB);
+  
+  if($gifts_dbh){
+    print "Got connection to GiftsDB \n";
+  }
+  
+  if($protfeature_dbh){
+    print "Got connection to ProtfeatureDB \n";
+  }
+  
+  my $insert_container = build_insert_sql($gifts_dbh, $protfeature_dbh);
+
+  
+  my $rows_inserted = 0;
+  my @failed_stmts = [];
+  unless($dryrun){
+
+  foreach my $insert_stmt (@$insert_container) {
```

Comment:
For future readability, could the indentation of this with respect to the dryrun if statement be fixed please.